src/Tests/Microsoft.NET.Build.Tests/GlobalPropertyFlowTests.cs

```diff
@@ -0,0 +1,276 @@
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public void TestGlobalPropertyFlowToExe(bool passSelfContained, bool passRuntimeIdentifier)
+        {
+            _referencedProject.IsExe = true;
+
+            var testAsset = Build(passSelfContained, passRuntimeIdentifier, identifier: passSelfContained.ToString() + "_" + passRuntimeIdentifier);
+
+            bool buildingSelfContained = passSelfContained || passRuntimeIdentifier;
+
+            ValidateProperties(testAsset, _testProject, expectSelfContained: buildingSelfContained, expectRuntimeIdentifier: buildingSelfContained);
+            ValidateProperties(testAsset, _referencedProject, expectSelfContained: buildingSelfContained, expectRuntimeIdentifier: buildingSelfContained);
+        }
+
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public void TestGlobalPropertyFlowToExeWithSelfContainedFalse(bool passSelfContained, bool passRuntimeIdentifier)
+        {
+            _referencedProject.IsExe = true;
+            _referencedProject.AdditionalProperties["SelfContained"] = "false";
+
+            string identifier = passSelfContained.ToString() + "_" + passRuntimeIdentifier;
+
+            if (!passSelfContained && passRuntimeIdentifier)
+            {
+                //  This combination results in a build error because it ends up being a self-contained Exe referencing a framework dependent one
+                var testAsset = _testAssetsManager.CreateTestProject(_testProject, identifier: identifier);
+
+                new DotnetBuildCommand(testAsset, "-r", EnvironmentInfo.GetCompatibleRid())
+                    .Execute()
+                    .Should()
+                    .Fail()
+                    .And
+                    .HaveStdOutContaining("NETSDK1150");
+            }
+            else
+            {
+
+                var testAsset = Build(passSelfContained, passRuntimeIdentifier, identifier: identifier);
+
+                bool buildingSelfContained = passSelfContained || passRuntimeIdentifier;
+
+                ValidateProperties(testAsset, _testProject, expectSelfContained: buildingSelfContained, expectRuntimeIdentifier: buildingSelfContained);
+                //  SelfContained will only flow to referenced project if it's explicitly passed in this case
+                ValidateProperties(testAsset, _referencedProject, expectSelfContained: passSelfContained, expectRuntimeIdentifier: buildingSelfContained);
+            }
+        }
+
+        [Theory]
+        [InlineData(true, true)]
+        [InlineData(true, false)]
+        [InlineData(false, true)]
+        [InlineData(false, false)]
+        public void TestGlobalPropertyFlowToLibraryWithRuntimeIdentifier(bool passSelfContained, bool passRuntimeIdentifier)
```

Comment:
Any special cases to consider when you have an EXE that reference multiple projects where the .csproj defines `SelfContained` or `RuntimeIdentifier` values and then build from the commandline with the `-r` or `--self-contained`?

Reply:
I don't have any tests where multiple projects are referenced, but I don't think we really need that.  There are tests where the referenced project defines `SelfContained` or `RuntimeIdentifier`.